### PR TITLE
new fields for resource req and limit 

### PR DESCRIPTION
**Pull request by kinarashah**

### CODE REVIEW

File: apis/cluster.cattle.io/v1/types.go [hunk 1]
```diff
@@ -219,6 +221,8 @@ type ClusterNode struct {
 	v1.Node
 	NodeName    string
 	ClusterName string
+	Requested   v1.ResourceList `json:"requested,omitempty"`
```
Comment (alena1108) [inline apis/cluster.cattle.io/v1/types.go]: Would it be possible to make it a part of the NodeSpec? Along with the Allocatable/Capacity

File: apis/cluster.cattle.io/v1/types.go [hunk 2]
```diff
@@ -216,9 +218,33 @@ type AuthConfig struct {
...
-	ClusterName string
+	NodeName        string
+	ClusterName     string
+	metav1.TypeMeta `json:",inline"`
+	// Standard object’s metadata. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the desired behavior of the cluster node. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status
+	Spec v1.NodeSpec `json:"spec,omitempty"`
```
Comment (alena1108) [inline apis/cluster.cattle.io/v1/types.go]: I guess you don't need `Spec`. You can just define the type?

File: apis/cluster.cattle.io/v1/types.go [hunk 3]
```diff
@@ -216,9 +218,33 @@ type AuthConfig struct {
...
+	// Most recent observed status of the cluster. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status
+	Status ClusterNodeStatus `json:"status"`
+}
+
+type ClusterNodeStatus struct {
+	Capacity        v1.ResourceList        `json:"capacity,omitempty"`
```
Comment (alena1108) [inline apis/cluster.cattle.io/v1/types.go]: You should embed the v1.NodeStatus, and add 2 new fields

File: apis/cluster.cattle.io/v1/types.go [hunk 4]
```diff
@@ -216,9 +218,33 @@ type AuthConfig struct {
 }
 
 type ClusterNode struct {
-	v1.Node
-	NodeName    string
-	ClusterName string
+	NodeName        string
```
Comment (alena1108) [inline apis/cluster.cattle.io/v1/types.go]: can you move nodename, clustername down? so k8s native fields come first

**Comment by alena1108** [comment]
@kinarashah some build failures regarding go generate, could you check? a lot of stuff seems to be missing in zz too

**Comment by alena1108** [comment]
@kinarashah we should also set requested/limit on a clusterstatus in addition to clusternodestatus

**Comment by kinarashah** [comment]
closing this, had a bunch of files deleting unexpectedly with go generate. #39

**Pull request closed without merging**